src/python/WMCore/MicroService/MSPileup/MSPileupTasks.py

```diff
@@ -43,6 +39,23 @@ def __init__(self, dataManager, logger, rucioAccount, rucioClient):
         self.rucioClient = rucioClient
         self.report = MSPileupReport()
 
+    def pileupSizeTask(self):
+        """
+        Execute pileup size update task
+        """
+        # get pileup sizes and update them in DB
+        spec = {}
+        docs = self.mgr.getPileup(spec)
+        rucioAuthUrl = self.rucioClient.rucioParams['auth_host']
+        rucioHostUrl = self.rucioClient.rucioParams['rucio_host']
+        rucioToken = getRucioToken(rucioAuthUrl, self.rucioAccount)
+        containers = [r['pileupName'] for r in docs]
+        datasetSizes = getPileupContainerSizesRucio(containers, rucioHostUrl, rucioToken)
+        for doc in docs:
+            pileupSize = datasetSizes.get(doc['pileupName'], 0)
```

Comment:
I see you don't have any error handling here (in case the HTTP request fails), but I would say this is a good enough commitment for the moment, given that it will be retried again in the next cycle.

Reply:
Fair enough, I don't have specific preferences here. I can easily add try/except block though, please clarify your view on this.

Reply:
I added try/except block and committed the changes.